public/python/model_checker.py

```
@@ -68,8 +68,33 @@ def check_dependencies():
+    Attempts ``get_fallback_schema`` first and then falls back to a
+    generic regex based extraction. If all else fails ``IFC4`` is used.
+    """
+
+    if not version:
+        return "IFC4"
+
+    try:
+        candidate = get_fallback_schema(version)
+        if candidate != version:
+            return candidate
+    except Exception:
```

Pylint (3.3.7)

[warning] Catching too general exception Exception

(W0718)